anvill/src/Analysis/CrossReferenceResolver.cpp

```diff
@@ -650,12 +646,16 @@ CrossReferenceResolverImpl::ResolveCall(llvm::CallInst *call) {
 // Try to resolve `val` as a cross-reference.
 ResolvedCrossReference
 CrossReferenceResolverImpl::ResolveValue(llvm::Value *val) {
+  auto it = xref_cache.find(val);
```

Comment:
OK I think I understand; at a higher level you check the cache; though when we're resolving constants, we always recursively call `ResolveConstant`, so now we're losing out there. I think we want to move this back into `ResolveConstant` and `ResolveInstruction`.